pkg/controller/storagecluster/storagecluster.go

```diff
@@ -226,6 +240,11 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) waitingForMigrationApproval(cluster *corev1.StorageCluster) bool {
+	approved, err := strconv.ParseBool(cluster.Annotations[constants.AnnotationMigrationApproved])
+	return err == nil && !approved
```

Comment:
In normal scenario, we need to return `false` so it can proceed. 
So when the annotation does not exist, the err will be NotNil. This function will return false and the reconcile will proceed as expected.